syntax = "proto2";

import "protos/model.proto";
import "protos/hyperparams.proto";
import "protos/frcnn.proto";
import "protos/post_process.proto";


message WSODModel {
  extend Model {
    optional WSODModel ext = 1057;
  }

  // FRCNN configs.
  optional FRCNN frcnn_options = 1;

  // Path to the detection vocabulary file.
  optional string vocabulary_file = 11;

  // Hyperparams of all the FC layers.
  optional Hyperparams fc_hyperparams = 12;

  // OICR iterations.
  optional int32 oicr_iterations = 21 [default = 0];

  // OICR IoU threshold.
  optional float oicr_iou_threshold = 22 [default = 0.5];

  // OICR start step.
  optional int32 oicr_start_step = 23 [default = 0];

  // Config of the MIDN post processor.
  optional PostProcess midn_post_process = 31;

  // Config of the OICR post processor.
  optional PostProcess oicr_post_process = 32;

  // Resolutions at inference time.
  repeated int32 eval_min_dimension = 34;

  // If true, use proba_r_given_c as oicr_score_0.
  optional bool oicr_use_proba_r_given_c = 36 [default = true];

  // MIDN loss weight.
  optional float midn_loss_weight = 91 [default = 1.0];

  // OICR loss weight.
  optional float oicr_loss_weight = 92 [default = 1.0];
}
